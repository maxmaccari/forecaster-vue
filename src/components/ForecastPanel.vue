<template>
  <div>Forecast Here</div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  async setup() {
    await new Promise((resolve, reject) => {
      setTimeout(() => reject({ message: 'my error' }), 1000)
    })

    return {}
  },
})
</script>
