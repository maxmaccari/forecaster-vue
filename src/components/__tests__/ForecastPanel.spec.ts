import { shallowMount } from '@vue/test-utils'
import ForecastPanel from '../ForecastPanel.vue'

describe('ForecastPanel', () => {
  it('renders the component properly', () => {
    const location = 'berlin'
    const wrapper = shallowMount(ForecastPanel, {
      props: {
        location,
        forecast: {}
      },
      global: {
        components: {
          VIcon: {},
          RouterLink: {}
        },
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
