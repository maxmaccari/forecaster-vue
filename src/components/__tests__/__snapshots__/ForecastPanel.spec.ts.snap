// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastPanel renders the component properly 1`] = `
<div class="w-full max-w-4xl p-2 text-indigo-900 bg-gray-100 sm:p-4 md:p-6">
  <div class="flex flex-wrap"><span class="mr-4 text-6xl leading-none align-text-top">11º </span>
    <div class="flex flex-col mt-1 text-indigo-800">
      <div class="border-gray-400 sm:border-l sm:pl-2">27 Sunday</div>
      <div class="flex items-center mt-1">
        <div>
          <v-icon-stub file="021-cloud" class="w-8"></v-icon-stub>
        </div>
        <div class="flex flex-col ml-2 text-xs"><span class="capitalize">overcast clouds</span><span> Fees like: 10º </span></div>
      </div>
    </div>
    <div class="flex flex-col w-full mt-1 border-gray-400 sm:border-l sm:pl-2 sm:ml-4 sm:w-auto">
      <div><span class="font-light">Min</span>: 9.44º </div>
      <div><span class="font-light">Max</span>: 12.22º </div>
    </div>
  </div>
  <div class="flex flex-col sm:flex-row sm:justify-between">
    <div>
      <div class="text-3xl sm:text-4xl">Berlin</div><a class="sm:block sm:w-48 change-btn" to="[object Object]">
        <v-icon-stub file="pencil-icon" class="inline-block w-4 fill-current"></v-icon-stub><span>Change City</span>
      </a>
      <div class="mt-1 text-sm text-gray-400 sm:text-xs"> Data fecthed from OpenWeather® API </div>
    </div>
    <div class="mt-2 sm:max-w-md sm:flex-grow sm:mt-4">
      <div class="sm:text-sm">Next 24 hours summary</div>
      <div class="flex flex-wrap">
        <div class="flex w-1/2 mt-2 sm:w-1/3 sm:mt-2">
          <div>
            <v-icon-stub file="021-cloud" class="w-8"></v-icon-stub>
          </div>
          <div class="ml-2">
            <div class="text-3xl font-bold leading-none sm:text-2xl">11º </div>
            <div class="text-lg font-light sm:text-base sm:mt-1">5:00 PM</div>
            <div class="text-sm text-gray-600 sm:text-xs">100% cloudness </div>
          </div>
        </div>
        <div class="flex w-1/2 mt-2 sm:w-1/3 sm:mt-2 border-gray-400 sm:border-l sm:pl-2">
          <div>
            <v-icon-stub file="021-cloud" class="w-8"></v-icon-stub>
          </div>
          <div class="ml-2">
            <div class="text-3xl font-bold leading-none sm:text-2xl">10º </div>
            <div class="text-lg font-light sm:text-base sm:mt-1">11:00 PM</div>
            <div class="text-sm text-gray-600 sm:text-xs">52% cloudness </div>
          </div>
        </div>
        <div class="flex w-1/2 mt-2 sm:w-1/3 sm:mt-2 border-gray-400 sm:border-l sm:pl-2 mt-4">
          <div>
            <v-icon-stub file="021-sun" class="w-8"></v-icon-stub>
          </div>
          <div class="ml-2">
            <div class="text-3xl font-bold leading-none sm:text-2xl">14º </div>
            <div class="text-lg font-light sm:text-base sm:mt-1">5:00 AM</div>
            <div class="text-sm text-gray-600 sm:text-xs">0% cloudness </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="mt-4 md:mt-8">
    <div class="text-lg">Next Week</div>
    <div class="grid grid-flow-row grid-cols-2 gap-2 mt-2 sm:gap-3 sm:grid-cols-3 md:grid-cols-6 md:gap-2"><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">27 Sunday</div>
        <v-icon-stub file="021-cloud" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">overcast clouds</div>
        <div class="text-xs capitalize md:mt-1">100% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 10º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">28 Monday</div>
        <v-icon-stub file="021-cloud" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">scattered clouds</div>
        <div class="text-xs capitalize md:mt-1">32% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 10º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">29 Tuesday</div>
        <v-icon-stub file="021-cloud" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">broken clouds</div>
        <div class="text-xs capitalize md:mt-1">77% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 12º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">30 Wednesday</div>
        <v-icon-stub file="021-sun" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">clear sky</div>
        <div class="text-xs capitalize md:mt-1">7% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 12º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">1 Thursday</div>
        <v-icon-stub file="021-cloudy-1" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">broken clouds</div>
        <div class="text-xs capitalize md:mt-1">78% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 13º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a><a to="[object Object]" class="relative p-2 text-indigo-100 bg-indigo-900 cursor-pointer hover:bg-indigo-800">
        <div class="md:text-sm">2 Friday</div>
        <v-icon-stub file="021-cloud" class="w-20 mx-auto mt-1 md:w-16 md:mt-2"></v-icon-stub>
        <div class="mt-1 text-xl font-bold md:text-lg md:mt-2">37º</div>
        <div class="text-sm capitalize md:text-xs md:mt-2">overcast clouds</div>
        <div class="text-xs capitalize md:mt-1">99% cloudiness </div>
        <div class="text-xs capitalize md:mt-1">Min: 13º</div>
        <v-icon-stub file="circle-right" class="absolute bottom-0 right-0 w-5 mb-2 mr-2 fill-current md:w-4"></v-icon-stub>
      </a></div>
  </div>
</div>
`;
